packages/client/hmi-client/src/services/data.ts

```diff
@@ -158,96 +192,137 @@ const getAssets = async (
 	};
 
 	// apply facet filters
-	if (searchParam && searchParam.filters && searchParam.filters.clauses.length > 0) {
-		// modelSearchParam currently represent facets filters that can be applied
-		//  to further refine the list of models
-
-		// a special facet related to ontology/DKG concepts needs to be transformed into
-		//  some form of filters that can filter the list of models.
-		// In this case, each concept has an associated list of model IDs that can be used to filter models
-		//  so, we need to map the facet filters from field "concepts" to "id"
-
-		// Each clause of 'concepts' should have another corresponding one with 'id'
-		const curies = [] as ClauseValue[];
-		const idClauses = [] as Clause[];
-		searchParam.filters.clauses.forEach((clause) => {
-			if (clause.field === CONCEPT_FACETS_FIELD) {
-				const idClause = cloneDeep(clause);
-				idClause.field = 'id';
-				const clauseValues = [] as ClauseValue[];
-				idClause.values.forEach((conceptNameOrCurie) => {
-					// find the corresponding model IDs
-					if (conceptFacets !== null) {
-						const matching = conceptFacets.results.filter(
-							(conceptResult) =>
-								conceptResult.name === conceptNameOrCurie ||
-								conceptResult.curie === conceptNameOrCurie
-						);
-						// update the clause value by mapping concept/curie to model id
-						clauseValues.push(...matching.map((m) => m.id));
-						curies.push(...matching.map((m) => m.curie));
+	if (searchParam) {
+		// Filtering for model/dataset data
+		if (searchParam && searchParam.filters && searchParam.filters.clauses.length > 0) {
```

Comment:
You can use lodash [`isEmpty()`](https://lodash.com/docs/4.17.15#isEmpty)
```js
import { isEmpty } from 'lodash';
...
if (!isEmtpy(searchParam?.filters?.clauses) {
    ...
```

Reply:
Awesome! Ill fix this
Thanks for the tip I will have to keep this library in mind

Reply:
Unfortunately I still need the first two arguments for the code inside to not freak out thinking the object could be empty.